build_test.go

```diff
@@ -68,49 +61,106 @@ func testBuild(t *testing.T, context spec.G, it spec.S) {
 				Entries: []packit.BuildpackPlanEntry{},
 			},
 			Layers: packit.Layers{Path: layersDir},
-		})
-		Expect(err).NotTo(HaveOccurred())
+		}
+	})
 
-		Expect(result).To(Equal(packit.BuildResult{
-			Plan: packit.BuildpackPlan{
-				Entries: nil,
-			},
-			Layers: nil,
-			Launch: packit.LaunchMetadata{
-				Processes: []packit.Process{
-					{
-						Type:    "web",
-						Command: "poetry run some-script",
-						Default: true,
-					},
-				},
-			},
-		}))
+	it.After(func() {
+		Expect(os.RemoveAll(layersDir)).To(Succeed())
+		Expect(os.RemoveAll(cnbDir)).To(Succeed())
+		Expect(os.RemoveAll(workingDir)).To(Succeed())
 	})
 
-	context("failure cases", func() {
-		context("when the pyproject.toml parser returns an error", func() {
-			it.Before(func() {
-				pyProjectParser.ParseCall.Returns.Error = fmt.Errorf("some error")
-			})
+	context("with BP_POETRY_RUN_TARGET not set", func() {
+		it("returns a result that sets the 'poetry run' launch command", func() {
+			result, err := build(buildContext)
+			Expect(err).NotTo(HaveOccurred())
 
-			it("returns the error", func() {
-				_, err := build(packit.BuildContext{
-					WorkingDir: workingDir,
-					CNBPath:    cnbDir,
-					Stack:      "some-stack",
-					BuildpackInfo: packit.BuildpackInfo{
-						Name:    "Some Buildpack",
-						Version: "some-version",
-					},
-					Plan: packit.BuildpackPlan{
-						Entries: []packit.BuildpackPlanEntry{},
+			Expect(result).To(Equal(packit.BuildResult{
+				Plan: packit.BuildpackPlan{
+					Entries: nil,
+				},
+				Layers: nil,
+				Launch: packit.LaunchMetadata{
+					Processes: []packit.Process{
+						{
+							Type:    "web",
+							Command: "poetry run some-script",
+							Default: true,
+						},
 					},
-					Layers: packit.Layers{Path: layersDir},
+				},
+			}))
+
+			Expect(buffer.String()).To(ContainLines(
```

Comment:
As above, personally I don't think there is much value in unit testing the log lines. The assertions on the tests are sufficient to determine whether we are entering the correct path.